### PR TITLE
Remove flash errors from admin password recovery

### PR DESCRIPTION
**Pull request by jacobherrington**
It is possible for an attack to deduce admin account email addresses by
injecting email addresses into the password recovery view until the
receive a success message.

The solution in this commit is to remove those flash error messages from
that view entirely. Instead we will flash a generic success message when
the form submits.

This functionality will still exist for non-admin users who may be less
technically savvy and may need help remembering the email address they
used to sign up.

Before: 
image

After:
image

### CODE REVIEW

File: spec/features/admin/password_reset_spec.rb [hunk 1]
```diff
@@ -6,19 +6,37 @@
...
 
-  scenario 'allows a user to supply an email for the password reset' do
-    user = create(:user, email: 'foobar@example.com', password: 'secret', password_confirmation: 'secret')
-    visit spree.admin_login_path
-    click_link 'Forgot Password?'
-    fill_in 'Email', with: 'foobar@example.com'
-    click_button 'Reset my password'
-    expect(page).to have_text 'You will receive an email with instructions'
+  context 'when the an account with this email address exists' do
+    before { user = create(:user, email: 'foobar@example.com', password: 'secret', password_confirmation: 'secret') }
```
Comment (spaghetticode) [inline spec/features/admin/password_reset_spec.rb]: I think the variable assignent `user = ` can be removed ✂️

Comment (tvdeyen) [reply spec/features/admin/password_reset_spec.rb]: I would prefer a `let!(:user)` instead.

File: spec/features/admin/password_reset_spec.rb [hunk 2]
```diff
@@ -6,19 +6,37 @@
...
   end
 
-  scenario 'allows a user to supply an email for the password reset' do
-    user = create(:user, email: 'foobar@example.com', password: 'secret', password_confirmation: 'secret')
-    visit spree.admin_login_path
-    click_link 'Forgot Password?'
-    fill_in 'Email', with: 'foobar@example.com'
-    click_button 'Reset my password'
-    expect(page).to have_text 'You will receive an email with instructions'
+  context 'when the an account with this email address exists' do
```
Comment (kennyadsl) [inline spec/features/admin/password_reset_spec.rb]: `the` can be removed here, right?

File: spec/features/admin/password_reset_spec.rb [hunk 3]
```diff
@@ -6,19 +6,37 @@
...
+
+    scenario 'allow a user to supply an email for the password reset' do
+      visit spree.admin_login_path
+      click_link 'Forgot Password?'
+      fill_in_email
+      click_button 'Reset my password'
+      expect(page).to have_text 'you will receive an email with instructions'
+    end
+
+    scenario 'email field is required' do
```
Comment (kennyadsl) [inline spec/features/admin/password_reset_spec.rb]: Honestly, I don't see a lot of value in this spec. We are just checking that we wrote `required: true` in the view. Also, is that change required in this PR? I think we should keep it as simple as possible but maybe I'm missing something?

**Comment by kennyadsl** [comment]
Another thing: I don't think an attack can deduce **admin** emails at the moment since the email existence check is performed against all users, not only admin ones. I think that every user, even non-admin, can login/reset password with those backend authentication pages. The only information that an attack can deduce is if that email belongs to a user signed up on the store, which we are also allowing to do in the frontend.

I think this is still a good change but I would consider adding the same behavior in frontend as well. Users that forget the email used can try different emails and they will receive an email while the right one is entered.

File: spec/features/admin/password_reset_spec.rb (hunk 3 above)
Comment (jacobherrington) [reply spec/features/admin/password_reset_spec.rb]: I left this test because previously a blank submission would render an error because ActiveRecord is validating that field, and we were testing that functionality in this feature spec.

I made the choice to remove all errors from this page, including that blank field error, but I added `required: true` in the view so that, generally, the behavior submitting the form with a blank field is not possible. 

I also added this because I think it is foolish not to require a field that will result in a failing validation if it is left blank.

We can remove the test, but I thought it was a good idea to leave it there because it validates that a similar behavior still exists.

Thoughts?

**Comment by jacobherrington** [comment]
@kennyadsl That is a good point.

I will extend this behavior to the frontend password reset in another PR, or look into opportunities to address duplicated functionality. I saw a few things I could improve in this gem while making this change, but I'd like to get @aitbw PRs merged in before doing too much more.

File: spec/features/admin/password_reset_spec.rb (hunk 3 above)
Comment (kennyadsl) [reply spec/features/admin/password_reset_spec.rb]: I'm fine adding the `require: true` but personally, I'd remove this test as is or I'd add an expectation that a client-side validation error is displayed if the email is left blank.
